### ISSUE TITLE
Apparently the 1.4.8 launcher has a Trojan in it according to windows defender 

### ISSUE THREAD

**Issue by Ponchos9911** [remyroy/CDDA-Game-Launcher#434]
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. Windows]
 - Version [e.g. 10]

**Additional context**

**Comment by remyroy** [comment]
I will refer you to [the FAQ](https://github.com/remyroy/CDDA-Game-Launcher#my-antivirus-product-detected-the-launcher-as-a-threat-what-can-i-do):

> Poor antivirus products are known to detect the launcher as a threat and block its execution or delete the launcher. A simple workaround is to add the launcher binary in your antivirus whitelist or select the action to trust this binary when detected.

> If you are paranoid, you can always inspect the source code yourself and build the launcher from the source code. You are still likely to get false positives. There is little productive efforts we can do as software developers with these. We have [a nice building guide](https://github.com/remyroy/CDDA-Game-Launcher/blob/master/BUILDING.md) for those who want to build the launcher from the source code.

**Comment by michaelbradley91** [comment]
@remyroy : I checked this as well, and it genuinely seems to have Malware inside it.
I think something in your build process may have included it accidentally.

I built the installer from source, and it is considered safe by Windows Defender, but your distribution includes wacatac, a Trojan. https://howtofix.guide/trojanwin32-wacatac-cml/

Please could you fix this asap?
Thanks

**Comment by Ponchos9911** [comment]
I can understand having poor antivirus could do that but if i was able to have the 1.4.4 and 1.4.5 versions with out any issue where as on one of the other open issue's they say they have had the issue with other files that also sat they contain a Trojan etc: Trojan:Win32/Fuery.B!cl and Trojan:Win32/Fuerboos.C!cl but didn't have this issue with so i was just wondering how if you sorted the issue with those files or it could actually be something else.

**Comment by remyroy** [comment]
How do you know it's not a false positive?

**Comment by remyroy** [comment]
@michaelbradley91 If you built it yourself and you don't have any AV warning, you are already good and you don't need to download our releases.

**Comment by michaelbradley91** [comment]
I thought it wasn't a false positive because after building from source, uploading and then downloading again, Windows Defender thought it was okay. I also scanned the file built from source manually with Windows Defender and it was considered safe yesterday.

However, for some reason this morning Windows has decided it now does think it is a Trojan again!! :( Even scanning the same file as yesterday now shows it up as a Trojan so not sure what happened there.

So, sorry, it could be a false positive. I might look into the difference between 1.4.4 and 1.4.8 later and see if I can figure out why Windows Defender thinks it is now dangerous. I'll let you know if I figure out what it is.

Like others, I see this output:
![image](https://user-images.githubusercontent.com/1542506/78473825-7c469e00-773b-11ea-8323-ed84f76fafc8.png)

**Comment by michaelbradley91** [comment]
@remyroy : good news! 😃 It looks like the anti-virus is classifying it as a Trojan due to some side effect of using Python37 rather than Python36. Hopefully their security definitions will be updated soon.

In the meantime, if you released a version built with Python36, that should cover the issue at least until then. (Up to you - I'm happy with my own build either way).

That said, maybe my AV will consider the 36 version a Trojan tomorrow so I might be speaking too soon... hopefully not.

For the long version:
So I took a look over all the changes between 1.4.4 and 1.4.8 and naturally nothing there seemed like it should trip AV. I modified the source on my machine a bit so I could use all the same dependencies as 1.4.4 as well, with the exception of the rfc package. My version was eventually built with these dependencies:

```
alembic==1.4.2
altgraph==0.17
arrow==0.15.5
Babel==2.8.0
certifi==2020.4.5
chardet==3.0.4
cssselect==1.1.0
future==0.18.2
html5lib==1.0.1
idna==2.9
LEPL==5.1.3
lxml==4.5.0
Mako==1.1.2
Markdown==3.2.1
MarkupSafe==1.1.1
pefile==2019.4.18
PyInstaller==3.6
pylzma==0.5.0
pypiwin32==223
PyQt5==5.14.2
PyQt5-sip==12.7.2
python-dateutil==2.8.1
python-editor==1.0.4
python-slugify==1.2.6
pytz==2019.3
pywin32==227
pywin32-ctypes==0.2.0
pywinutils==0.2
rarfile==3.1
requests==2.23.0
six==1.14.0
SQLAlchemy==1.3.15
transifex-client==0.13.8
Unidecode==1.1.1
urllib3==1.25.8
webencodings==0.5.1
```

Or in other words, the requirements.txt looks like this:
```
SQLAlchemy
alembic
PyQt5
PyInstaller>=3.4
html5lib
lxml
cssselect
arrow
rarfile
pylzma
Babel
pypiwin32
pywinutils
Markdown
transifex-client
```

Which except for the missing rfc6266 package should be the same as 1.4.4. Even with this, when built with Python 37 according to virus total at least, Microsoft considers the binary a Trojan.

However, when built with Python 36, it is considered safe. I then just took 1.4.8's source, rebuilt that with **the only changing being to use Python36**, and again the exectuable was considered safe by Microsoft.

Anyway, I'm happy it is a false positive and sorry for my confusion earlier. 😕

**Comment by remyroy** [comment]
I have a good idea of the souce and the cause of these poor antivirus products showing a false positive like this one. There is [a long history](https://github.com/pyinstaller/pyinstaller/issues?q=is%3Aissue+virus+is%3Aclosed) related to this issue. I'm sorry to say that there is sincerely very little a reasonable software developer can do about it.

If you truely want to help about this, you should submit the launcher to each and every poor antivirus product's false positive process to clear it and do so each time there is a new version of either the launcher, PyInstaller or Python released. In my humble opinion, this process is too much time consuming for the potential result we may or may not get even if those companies cared about their product, their reputation or the well being of their users which is often not the case.

**Comment by Loveblanket** [comment]
I'm only using Windows Defender which has served me just fine and I can't even add the launcher as an exemption in the virus section, it erases the file as soon as it's installed. How do we "build our own". I have no idea how to do that.

**Comment by remyroy** [comment]
@Loveblanket Check out these articles:

https://support.microsoft.com/en-us/help/4028485/windows-10-add-an-exclusion-to-windows-security

https://deeparmorsupport.sparkcognition.com/hc/en-us/articles/360018015392-How-to-Whitelist-Microsoft-Windows-Defender

**Comment by biroborg** [comment]
The same issue happens with 1.4.10 since today:
Threat detected: Trojan:Win32/Occamy.C

**Comment by remyroy** [comment]
@biroborg Have you read [the FAQ](https://github.com/remyroy/CDDA-Game-Launcher#my-antivirus-product-detected-the-launcher-as-a-threat-what-can-i-do)?

**Comment by biroborg** [comment]
> 
> 
> @biroborg Have you read [the FAQ](https://github.com/remyroy/CDDA-Game-Launcher#my-antivirus-product-detected-the-launcher-as-a-threat-what-can-i-do)?

Yes, I've already added an exception for the file and posted just in case someone searches for the issue with the same version of CDDA.
